tests/Form/DataTransformer/OptionalDomainEmailTransformerTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Tests\Form\DataTransformer;
+
+use App\Form\DataTransformer\OptionalDomainEmailTransformer;
+use PHPUnit\Framework\TestCase;
+
+class OptionalDomainEmailTransformerTest extends TestCase
+{
+    const DOMAIN = 'systemli.org';
+
+    /**
+     * @dataProvider transformProvider
+     */
+    public function testTransform($input, $expected)
+    {
+        $this->assertEquals($expected, $this->getTransformer()->transform($input));
+    }
+
+    public function transformProvider()
+    {
+        return [
+            ['', ''],
+            [null, ''],
+            ['louis@systemli.org', 'louis'],
+            ['louis', 'louis'],
+        ];
+    }
+
+    /**
+     * @dataProvider reverseTransformProvider
+     */
+    public function testReverseTransform($input, $expected)
+    {
+        $this->assertEquals($expected, $this->getTransformer()->reverseTransform($input));
+    }
+
+    public function reverseTransformProvider()
+    {
+        return [
+            ['', ''],
+            [null, ''],
+            ['louis', 'louis@systemli.org'],
+            ['louis@example.org', 'louis@example.org'],
```

Comment:
Not sure whether we should use real adresses in the tests. I know that it has been like that in the past, but ...